Modules/EMCAL/src/CalibMonitoringTask.cxx

```diff
@@ -88,6 +104,14 @@ void CalibMonitoringTask::update(Trigger t, framework::ServiceRegistryRef)
       for (Int_t i = 0; i < hist_temp2->GetNbinsX(); i++) {
         for (Int_t j = 0; j < hist_temp2->GetNbinsY(); j++) {
           mBadChannelMapHisto->SetBinContent(i + 1, j + 1, hist_temp2->GetBinContent(i + 1, j + 1));
+          // check to see if the cell is in the emcal or dcal and fill appropriate hist
+          int cellID = geo->GetAbsCellIdFromCellIndexes(i, j);
+          if (cellID < 12288) {
+            mBadChannelMapEMCALHisto->Fill(hist_temp2->GetBinContent(i + 1, j + 1));
+          }
+          else{
+            mBadChannelMapDCALHisto->Fill(hist_temp2->GetBinContent(i + 1, j + 1));
+          }
```

Comment:
There is also a pointer to the bad channel map itself, you would not need the geometry functions, you know the tower ID. Iterating over the bad channel map is super fast, calculating the the cell indices is potentially expensive, a separate for loop makes sense for performance reasons.